src/Microsoft.Extensions.HealthChecks.Abstractions/IHealthCheck.cs

```diff
@@ -0,0 +1,23 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.HealthChecks
+{
+    /// <summary>
+    /// Represents a health check, which can be used to check the status of an external dependency, such as a database or backend service.
+    /// </summary>
+    public interface IHealthCheck
+    {
+        /// <summary>
+        /// Gets the name of the health check, which should indicate the component being checked.
+        /// </summary>
+        string Name { get; }
```

Comment:
This is a programmatic name right? (for serialization and configuration) vs for display

Reply:
Mostly, though it would be "displayed" via the JSON "/health" endpoint (when that's added).